summingbird-storm/src/main/scala/com/twitter/summingbird/storm/StormTopologyBuilder.scala

```diff
@@ -0,0 +1,157 @@
+
+case class StormTopologyBuilder(options: Map[String, Options], jobId: JobId, stormDag: Dag[Storm]) {
+  @transient private val logger = LoggerFactory.getLogger(classOf[StormTopologyBuilder])
+
+  def build(): StormTopology = {
+    val topologyBuilder = new TopologyBuilder
+    stormDag.nodes.foreach {
+      case summerNode: SummerNode[Storm] => scheduleSummerBolt(summerNode, topologyBuilder)
+      case flatMapNode: FlatMapNode[Storm] => scheduleFlatMapper(flatMapNode, topologyBuilder)
+      case sourceNode: SourceNode[Storm] => scheduleSpout(sourceNode, topologyBuilder)
+    }
+    topologyBuilder.createTopology()
+  }
+
+  /**
+   * Set storm to tick our nodes every second to clean up finished futures
+   */
+  private def tickConfig = {
+    val boltConfig = new BacktypeStormConfig
+    boltConfig.put(BacktypeStormConfig.TOPOLOGY_TICK_TUPLE_FREQ_SECS, java.lang.Integer.valueOf(1))
+    boltConfig
+  }
+
+  private def scheduleFlatMapper(node: FlatMapNode[Storm], topologyBuilder: TopologyBuilder) = {
+    val bolt: BaseBolt[Any, Any] = FlatMapBoltProvider(this, node).apply
+
+    val parallelism = getOrElse(node, DEFAULT_FM_PARALLELISM).parHint
+    val declarer = topologyBuilder.setBolt(getNodeName(node), bolt, parallelism).addConfigurations(tickConfig)
+    bolt.applyGroupings(declarer)
+  }
+
+  private def scheduleSpout(node: SourceNode[Storm], topologyBuilder: TopologyBuilder) = {
+    val nodeName = stormDag.getNodeName(node)
+    val (sourceParalleism, stormSpout) = SpoutProvider(this, node).apply
+    topologyBuilder.setSpout(nodeName, stormSpout, sourceParalleism)
+  }
+
+  private def scheduleSummerBolt(node: SummerNode[Storm], topologyBuilder: TopologyBuilder): Unit = {
+
+    val nodeName = getNodeName(node)
+
+    def sinkBolt[K, V](summer: Summer[Storm, K, V]): BaseBolt[_, _] = {
+      implicit val semigroup = summer.semigroup
+      implicit val batcher = summer.store.mergeableBatcher
+
+      type ExecutorKeyType = (K, BatchID)
+      type ExecutorValueType = (Timestamp, V)
+      type ExecutorOutputType = (Timestamp, (K, (Option[V], V)))
+
+      val anchorTuples = getOrElse(node, AnchorTuples.default)
+      val metrics = getOrElse(node, DEFAULT_SUMMER_STORM_METRICS)
+      val shouldEmit = stormDag.dependantsOf(node).size > 0
+
+      val builder = BuildSummer(this, node)
+
+      val ackOnEntry = getOrElse(node, DEFAULT_ACK_ON_ENTRY)
+      logger.info(s"[$nodeName] ackOnEntry : ${ackOnEntry.get}")
+
+      val maxEmitPerExecute = getOrElse(node, DEFAULT_MAX_EMIT_PER_EXECUTE)
+      logger.info(s"[$nodeName] maxEmitPerExecute : ${maxEmitPerExecute.get}")
+
+      val maxExecutePerSec = getOrElse(node, DEFAULT_MAX_EXECUTE_PER_SEC)
+      logger.info(s"[$nodeName] maxExecutePerSec : $maxExecutePerSec")
+
+      val storeBaseFMOp = { op: (ExecutorKeyType, (Option[ExecutorValueType], ExecutorValueType)) =>
+        val ((k, batchID), (optiVWithTS, (ts, v))) = op
+        val optiV = optiVWithTS.map(_._2)
```

Comment:
I think we use opt as the prefix everywhere else so may be use that here as val. i.e optVWithTS and optV.